lib/speed/time-collector.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+class TimeCollector {
+  constructor(
+    count = 25000, // Number, number of test iterations
+    startCount = 0, // number of first optimizing iterations
+    anomalyPercent = 5 // Number, maximum possible percent of anomalies
+  ) {
+    this.count = count;
+    this.startCount = startCount;
+    this.anomalyPercent = anomalyPercent;
+    this.bufferSize = this.getBufferSize(count, anomalyPercent);
+    this.intermediateValue = [];
+    this.intermediateValue.add = (i, ...argv) => {
+      if (!Math.random()) {
```

Comment:
`!Math.random()` why?

Reply:
When the function is tested, it is important to store the resulting value of the function so that the optimizing compiler does not optimize the function. I keep the values in the array "InterValue". In order for the array to not take up much memory, I add the result of the function only if "Math.random () === 0" (the probability of this is very small).